### PR TITLE
DELETE /api/comments/:comment_id - task #12

### PR DESCRIPTION
**Pull request by jothwas**
This PR includes:

- minor tweaks to query tests in last task adding further tests following feedback (Dave was happy for me to do it in this next task)
- adds a 204 test to ensure the specified comment is deleted
- adds 400 test for invalid comment_id
- adds 404 test for path that does not exist

### CODE REVIEW

File: app/app.test.js [hunk 1]
```diff
@@ -279,6 +281,16 @@ describe("app", () => {
...
+            })
+            .then(() => {
+              return request(app)
+                .get("/api/articles")
+                .expect(200)
+                .then(({ body: { articles } }) => {
+                  expect(articles).toBeSortedBy("created_at", {
+                    descending: true,
+                  });
+                });
```
Comment (dav3rid) [inline app/app.test.js:294]: I actually missed this earlier so apologies!
I can see that you're actually chaining tests together in `.then`s here - if they are all in a separate test block then jest will actually run them concurrently. This, if only slightly, will be intentionally slowing down the test suite.
You could always put them in a Promise.all to have them trigger simultaneously, but separate test blocks will achieve the same thing

File: app/app.test.js [hunk 2]
```diff
@@ -347,6 +347,14 @@ describe("app", () => {
...
             });
         });
+        test("status: 200 - accepts a topic query and filters values by that topic when that topic has no values but is valid", () => {
+          return request(app)
+            .get("/api/articles?topic=paper")
+            .expect(200)
+            .then(({ body: { articles } }) => {
+              expect(articles).toHaveLength(0);
+            });
+        });
```
Comment (dav3rid) [inline app/app.test.js:345]: 🎉

File: app/app.test.js [hunk 3]
```diff
@@ -502,6 +502,27 @@ describe("app", () => {
...
+            return db.query("SELECT * FROM comments WHERE comment_id = 1");
+          })
+          .then(({ rows }) => {
+            expect(rows).toHaveLength(0);
+          })
+          .then(() => {
+            return db.query("SELECT * FROM comments");
+          })
+          .then(({ rows }) => {
+            expect(rows).toHaveLength(17);
```
Comment (dav3rid) [inline app/app.test.js:527]: I like this test - in any then block where nothing async is happening, you can group together.
Eg
```
.then(({ rows }) => {
    expect(rows).toHaveLength(0);
    return db.query("SELECT * FROM comments");
 })
```

File: app/app.test.js (hunk 1 above)
Comment (jothwas) [reply app/app.test.js:294]: Great - will refactor these.

**Pull request merged**